### PR TITLE
Feat/create bookmark

### PR DESCRIPTION
**Pull request by VirginiaBalseiro**
# Create bookmark

- Added a new Bookmark component that uses bookmarks => `useBookmark` checks if Bookmarks already exists or creates it at `/bookmarks/index.ttl`
- Bookmarks is kept in a context to avoid drilling it down several layers since the Bookmark icon is inside container > row component since it is used in every operation (add, remove, display selected/unselected,etc).
- when clicking on the star icon, it checks if the bookmark already exists and if so, it does nothing [we will add delete functionality later], if not, it adds the bookmark.

# Checklist

- [x] All acceptance criteria are met.
- [ ] Relevant documentation, if any, has been written/updated.
- [ ] The changelog has been updated, if applicable.
- [x] Commits in this PR are minimal and [have descriptive commit messages](https://chris.beams.io/posts/git-commit/).

### CODE REVIEW

File: src/solidClientHelpers/bookmarks.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+}
+
+export async function addBookmark(bookmarkIri, bookmarks, fetch) {
+  const { dataset, iri } = bookmarks;
+  const bookmarkTitle = getResourceName(bookmarkIri);
+  const bookmark = defineThing(
+    {},
+    (t) => addUrl(t, rdf.type, "http://www.w3.org/2002/01/bookmark#Bookmark"),
+    (t) =>
+      addStringNoLocale(t, "http://purl.org/dc/terms/title", bookmarkTitle),
```
Comment (megoth) [inline src/solidClientHelpers/bookmarks.js:52]: There's also the terms vocabulary in the "rdf-namespaces" package, let's use it.

**Review by megoth — CHANGES_REQUESTED** [review]
One minor change, and it's good to go ^_^

File: components/bookmark/index.jsx [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+      setBookmarked(true); // optimistically set bookmarked to true so it doesn't take that long to display the item as bookmarked
+      setBookmarks(response);
+    }
+  };
+};
+
+const isBookmarked = (iri, dataset) => {
+  const listOfRecallsUrls = getUrlAll(
+    dataset,
+    "http://www.w3.org/2002/01/bookmark#recalls"
```
Comment (ajacksified) [inline components/bookmark/index.jsx:66]: This (and all property IRIs) should be an exported constant.

File: components/bookmark/index.jsx [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+  );
+  if (!listOfRecallsUrls) {
```
Comment (ajacksified) [inline components/bookmark/index.jsx:68]: The solid-client-js api looks like it [always returns an array](https://docs.inrupt.com/developer-tools/api/javascript/solid-client/modules/_thing_get_.html#geturlall); you should be able to skip this check and just use the `includes` below.

**Review by megoth — APPROVED** [review]

**Pull request merged**